W-14069450-botsessioninformation-se (#293)
mulesoft/docs-rpa: rpa-home/modules/ROOT/pages/automation-security-securesessions.adoc

= Secure Sessions

> In RPA automations, a bot simulates a human user. The bot logs in to a system, sees the screen, and can interact with the UI via key strokes and mouse actions. That way, the bot can even interact with legacy systems, that do not provide an API.

Reviewer: ```suggestion
In RPA automations, a bot simulates a human user. The bot logs in to a system, sees the screen, and can interact with the UI via keystrokes and mouse actions. The bot can even interact with legacy systems that do not provide an API.
```

> The bot logs in to the computer, where it is installed. But, of course, it does not sit on a chair in front of the desktop. Nevertheless, if you would enter the room where the computer is, you could see what the bot is doing: The screen would show the open applications, the mouse would  move around and typing would appear in forms. In MuleSoft RPA, we call this a _Desktop Session_. To prevent unauthorized data access or manipulation, you would need to guard the physical room.

Reviewer: ```suggestion
The bot logs in to the computer where it is installed. But, of course, it does not sit on a chair in front of the desktop. Nevertheless, if you enter the room where the computer is, you can see what the bot is doing: The screen shows the open applications, the mouse moves around and typing appears in forms. In MuleSoft RPA, we call this a _Desktop Session_. To prevent unauthorized data access or manipulation, you must secure the physical room.
```

> Or you could use what we call _Secure Sessions_.

Reviewer: ```suggestion
Or you can use what RPA calls _Secure Sessions_.
```

> Secure Sessions are hidden Desktop Sessions. They run in the background and only the bot can access them. An additional advantage of the Secure Sessions is, that the bot can multiply itsself to run several process instances in parallel.

Reviewer: This sounds familiar. See https://github.com/mulesoft/docs-rpa/pull/293/files#r1376615143

> To use Secure Sessions you must ensure that all prerequisites are met and all responsible persons are informed.

Reviewer: You defined these as partials but still need to include them?
https://github.com/mulesoft/docs-rpa/pull/293/files#r1376618297

Author: will be part of the Doc Space Redesign project

== General Considerations

> * At the same time, a bot can only run in Desktop or in Secure Session mode.

Reviewer: ```suggestion
* A bot either can run in Desktop or Secure Session mode.
```

Author: Replaced with 
A bot either can run in Desktop Session or Secure Session mode. You can change the mode while no configurations are deployed to the bot.
[While - because it refers to a time span]

> * At the same time, a bot can only run in Desktop or in Secure Session mode.

Reviewer: Is there a way to further group these bullets into logical sub-bullets or hanging paragraphs? Like session mode, others? They seem sort of random right now.

Author: This needs more thinking about... --> Postponed

> * You can not switch the session mode while configurations are deployed on the bot.

Reviewer: ```suggestion
* You can't switch the session mode while configurations are deployed on the bot.
```

> * Run configurations specify a session mode, too.

Reviewer: Does the user specify the session mode for a run configuration? Or does the run configuration do that?

Author: The user specifies the mode for a run configuration and selects a fitting bot in one of the next steps.

* You can only deploy configurations to a bot that is in the same session mode.

> * You can specify the login data for the bot and the maximum number of Secure Sessions that the bot may run in parallel in the RPA Bot Configurator or in the RPA Manager.

Reviewer: ```suggestion
* You can specify the login data for the bot and the maximum number of secure sessions that the bot can run in parallel in the RPA Bot Configurator or the RPA Manager.
```

> ** The maximum number of Secure Sessions set and enabled in the RPA Bot Configurator overwrites the same setting of RPA Manager.

Reviewer: ```suggestion
+
The maximum number of secure sessions specified in the RPA Bot Configurator overwrites the same setting in RPA Manager.
```

Reviewer: Previously, I had a question about the wording of something similar (limits).

Reviewer: (Removed the single sub-bullet, per style)

* Desktop Sessions:

> ** The specified user must be logged in to this session to deploy a run configuration. 

Reviewer: should "this session" be "the current session" or "the same session"?

Author: The user must be logged in which  creates a session. This session is then overtaken by the bot.
This is already described somewhere...
--> Postponed

> ** Do not close or minimize the console session while the bot executes a process.

Reviewer: Our style is to make bullet list items parallel. This bullet is an imperative instruction, but the previous one is just a requirement. Let me know if this is unclear.

* Secure Sessions:

> ** The higher the number of secure sessions the lower the performance of the bot. Too much load can cause process slowdown or disruptions. Check the system data and the system requirements documentation to determine the maximum number of sessions that the bot can safely execute in parallel.

Reviewer: ```suggestion
** The higher the number of secure sessions, the lower the performance of the bot. Too much load can cause process slowdown or disruptions. Check the system data and the system requirements documentation to determine the maximum number of sessions that the bot can safely execute in parallel.
```

Reviewer: Suggest an alternative way to word this - something like "The number of secure sessions can negatively affect bot performance. 

How much is "too much load"? 

Can you link to the requirements document here?

> ** Before deploying a run configuration on a bot that has only one secure session licensed, do not disconnect the session, but sign out the specified user. 

Reviewer: See previous comment on this: https://github.com/mulesoft/docs-rpa/pull/293/files#r1376641014
It looks like you defined the partial, but didn't include it here?

> ** Do not sign in as the specified user to a computer where a bot with one licensed secure session is running a process.

Reviewer: Duplicated in partials file - add include here?

== Next Steps

> . In the documentation, check the System Requirements.

Reviewer: link

Author: linked in See Also
--> postponed to RPA Doc Space Reddisign

> . On the computer where you want to run automations and where the bot therefore is installed, check the system data:

Reviewer: ```suggestion
. On the computer where you want to run automations, where the bot is installed, check the system data:
```

+
* Operating system
* RAM

> * Terminal Server software

Reviewer: Terminal server?

Author: it's a name

* Login data for the bot
* RPA Bot Configurator settings

> . If you want to use default session settings for the run configuration, set the bot login data and the maximum number of parallel Secure Sessions in the Bot Configurator. 

Reviewer: Global - check to see if you can lowercase secure sessions, except when referring to the feature.

Author: uppercased all, because it'S always the feature

. In RPA Manager, enter the relevant data or a contact person in the description of the bot in the Bots view of the Bot Management module.

> .. If the bot should use Secure Sessions and the RPA Bot Configurator does not contain this setting, set the maximum number of parallel Secure Sessions that the bot may use via the Secure Session Assignment view.

Reviewer: ```suggestion
+
If the bot should use Secure Sessions and the RPA Bot Configurator does not contain this setting, set the maximum number of parallel Secure Sessions that the bot can use via the Secure Session Assignment view.
```

Reviewer: I noticed that you had a single sub-step. This should just be a hanging paragraph. Our style is to not have single un/ordered list items.

> . When creating a run configuration in the Process Automation module of RPA Manager, keep the general considerations in mind. 

Reviewer: link? 

Should this be something like "Review the General Considerations before..."?

== See Also

> * xref:rpa-bot::hardware-software-requirements.adoc[RPA Bot - Hardware and Software Requirements]

Reviewer: Global - avoid hard-coded link text if possible.

Or you can use what MuleSoft RPA calls _Secure Sessions_.

Secure Sessions are hidden Desktop sessions that run in the background and are accessible only to the bot. An additional advantage of using Secure Sessions is that the bot can multiply itself to run several process instances in parallel.

To use Secure Sessions you must ensure that all prerequisites are met and all responsible persons are informed.

== General Considerations

* A bot either can run in Desktop Session or Secure Session mode. You can change the mode while no configurations are deployed to the bot.
* You can't switch the session mode while configurations are deployed on the bot.
* Run configurations specify a session mode, too.
* You can only deploy configurations to a bot that is in the same session mode.
* You can specify the login data for the bot and the maximum number of Secure Sessions that the bot can run in parallel in the RPA Bot Configurator or in the RPA Manager.
** The maximum number of Secure Sessions specified in the RPA Bot Configurator limits overwrites the same setting in RPA Manager.
+

> Enabling the same setting via the RPA Bot Configurator, limits but not not overwrites this setting.

Reviewer 2: ```suggestion
Enabling the same setting via the RPA Bot Configurator limits but does not overwrite this setting.
```

Author: Replaced with Hanna's suggestion from another file (organizing these informations and implementimng them as partials will be part of the Doc Space Redesign project.)

* Desktop Sessions:
** The specified user must be logged in to this session to deploy a run configuration. 
** The console session must remain open and unminimized while the bot executes a process.
* Secure Sessions:
** The higher the number of Secure Sessions the lower the performance of the bot. Too much load can cause process slowdown or disruptions. Check the system data and the system requirements documentation to determine the maximum number of sessions that the bot can safely execute in parallel.

> ** Before deploying a run configuration on a bot that has only one Secure Session licensed, do not disconnect the session, but sign out the specified user. 

Reviewer 2: ```suggestion
** Before deploying a run configuration on a bot that has only one Secure Session licensed, do not disconnect the session but do sign out the specified user. 
```

Author: Replaced all occurrences with suggestions from Hanna
